Code.Rmd

```diff
@@ -7,19 +7,32 @@ source("common.R")
 # R code {#r}
 
 The first principle of using a package is that all R code goes in `R/`.
-In this chapter, you'll learn about the `R/` directory, my recommendations for organising your functions into files, and some general tips on good style.
-You'll also learn about some important differences between functions in scripts and functions in packages.
+In this chapter, you'll learn about the `R/` directory, our recommendations for organising your functions into files, and some general tips on good style.
+You'll also learn some important differences between defining a function in a script and defining it in a package.
 
 ## R code workflow {#r-workflow}
 
-The first practical advantage to using a package is that it's easy to re-load your code.
-You can either run `devtools::load_all()`, or in RStudio press __Ctrl/Cmd + Shift + L__,  which also saves all open files, saving you a keystroke.
+As your use of R becomes more sophisticated, it's common to start to write your own R functions.
+If a function is only used in one place, you probably define it right there.
+But if you've bothered to write a function, it's likely you want to reuse it in multiple places: within one script, across multiple scripts, or even across multiple projects.
+This is *exactly* what an R package is for!
+
+Without package technology, you probably collect these function definitions in one or more dedicated `.R` files and then `source()` them as needed.
+Typically these functions co-evolve with the code where you use them, i.e. your analysis code, your Shiny apps, or your R Markdown reports.
+If you use these functions across multiple projects, you also face the uncomfortable dilemma of where to define them and whether to have multiple, slightly different copies of this code lying around.
```

Comment:
This feels like a second point, that is:

* worth making
* possibly doesn't belong *right here* (but it sort of does)

Reply:
Although it begs the question of how you manage your local package installs, versioning etc.